bluebrain/repo-bluebrain/packages/py-region-grower/package.py

```diff
@@ -31,5 +33,7 @@ class PyRegionGrower(PythonPackage):
     depends_on("py-pandas@1.5.3:", type=("build", "run"))
     depends_on("py-tqdm@4.28.1:", type=("build", "run"))
     depends_on("py-voxcell@3.1.1:3", type=("build", "run"))
-    depends_on("py-dask-mpi@2022.4:", type=("build", "run"))
-    depends_on("py-mpi4py@3.1.1:", type=("build", "run"))
+
+    depends_on("py-dask-mpi@2022.4:", type=("build", "run"), when="+mpi")
+    depends_on("py-mpi4py@3.1.1:", type=("build", "run"), when="+mpi")
+    depends_on("mpi", type=("build", "run"), when="+mpi")
```

Comment:
Hi @matz-e 
I'm not sure about this `variant` part.
Here is the behavior I would like:
- if I use `module load py-region-grower` or `module load py-region-grower +mpi` it should load the MPI dependencies.
- if I use `module load py-region-grower -mpi` it should NOT load the MPI dependencies.
It it correct like this?

Also, should I write something more specific like `depends_on("hpe-mpi", ...)` instead of just `depends_on("mpi", ...)`